pkg/chunkenc/stats.go

```diff
@@ -0,0 +1,19 @@
+package chunkenc
+
+type StatsContext interface {
```

Comment:
Isn't the context also needed to communicate stops e.g. via `ctx.Done()`?

Reply:
Not used for stats no, It's just used as a transport.

Reply:
Couple of questions here.
1. I only see fakeStatsContext and noopStatsContext here. Which `StatsContext` should be used for real?
2. Should we have some adapter function that takes `contex.Context` and returns `StatsContext`?

Basically I'm trying to think how the caller of this package create a valid `StatsContext` instance ?

Reply:
Ah. You have `stats.FromContext()`. Never mind Ignore my previous comment.